### PR TITLE
doc: legacy withdrawal cell

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

**Review by Flouse — DISMISSED** [review]

File: docs/deposit_and_withdrawal.md [hunk 1]
```diff
@@ -94,9 +96,53 @@ struct WithdrawalLockArgs {
...
+    // layer1 lock to withdraw after challenge period
+    owner_lock_hash: Byte32,
+    // layer1 lock to receive the payment, must exists on the chain
+    payment_lock_hash: Byte32,
+}
+```
+
+We have optional fields in the withdrawal cell's args:
+
+* owner lock - If users submit withdrawal request with a owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they'are finalized. Users are no need to manually unlock layer-1 withdrawal cells.
```
Comment (Flouse) [inline docs/deposit_and_withdrawal.md]:
```suggestion
* owner lock - If users submit withdrawal request with an owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they are finalized. Users don't need to manually unlock layer-1 withdrawal cells.
```

File: docs/deposit_and_withdrawal.md [hunk 2]
```diff
@@ -94,9 +96,53 @@ struct WithdrawalLockArgs {
...
+    sell_amount: Uint128,
+    sell_capacity: Uint64,
+    // layer1 lock to withdraw after challenge period
+    owner_lock_hash: Byte32,
+    // layer1 lock to receive the payment, must exists on the chain
+    payment_lock_hash: Byte32,
+}
+```
+
+We have optional fields in the withdrawal cell's args:
```
Comment (Flouse) [inline docs/deposit_and_withdrawal.md]:
```suggestion
We have **optional** fields in the withdrawal cell's args:
```

File: docs/deposit_and_withdrawal.md [hunk 3]
```diff
@@ -94,9 +96,53 @@ struct WithdrawalLockArgs {
...
+
+* owner lock - If users submit withdrawal request with a owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they'are finalized. Users are no need to manually unlock layer-1 withdrawal cells.
+* withdrawal_to_v1 - This field only works when `owner lock` exist, if `withdrawal_to_v1` is exist and the value is `1`, which means the withdrawal is a fast withdraw to v1 Godwoken, the withdraw can be instantly processed, and the assets will be migrated to v1 Godwoken.
```
Comment (Flouse) [inline docs/deposit_and_withdrawal.md]:
```suggestion
* withdrawal_to_v1 - This field only works when `owner lock` exist, if `withdrawal_to_v1` is exist and the value is `1`, which means the withdrawal is a fast withdrawal to Godwoken v1. A fast withdrawal from v0 to v1 can be instantly processed, and the assets will be migrated to Godwoken v1.
```

File: docs/deposit_and_withdrawal.md [hunk 4]
```diff
@@ -94,9 +96,53 @@ struct WithdrawalLockArgs {
...
+* manually withdrawl - If `owner lock` do not exist, users must manually unlock the legacy withdrawal cell after it finalized, and user must provides an input cell in the unlocking transaction that it's `lock hash` is equals to withdrawal lock args' `owner_lock_hash`.
```
Comment (Flouse) [inline docs/deposit_and_withdrawal.md]:
```suggestion

### **Note**:
manually withdrawl - If `owner lock` do not exist, users must manually unlock the legacy withdrawal cell after it is finalized, and user must provides an input cell in the unlocking transaction that its `lock hash` is equals to withdrawal lock args' `owner_lock_hash`.
```

File: docs/deposit_and_withdrawal.md [hunk 5]
```diff
@@ -94,9 +96,55 @@ struct WithdrawalLockArgs {
...
+}
+```
+
+We have **optional** fields in the withdrawal cell's args:
+
+* owner lock - If users submit withdrawal request with an owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they are finalized. Users don't need to manually unlock layer-1 withdrawal cells.
+* withdrawal_to_v1 - This field only works when `owner lock` exist, if `withdrawal_to_v1` is exist and the value is `1`, which means the withdrawal is a fast withdrawal to Godwoken v1. A fast withdrawal from v0 to v1 can be instantly processed, and the assets will be migrated to Godwoken v1.
+
+### **Note**:
+manually withdrawl - If `owner lock` do not exist, users must manually unlock the legacy withdrawal cell after it is finalized, and user must provides an input cell in the unlocking transaction that its `lock hash` is equals to withdrawal lock args' `owner_lock_hash`.
```
Comment (ShookLyngs) [inline docs/deposit_and_withdrawal.md]: do not exist -> does not exist

**Review by ShookLyngs — CHANGES_REQUESTED** [review]

**Review by Flouse — APPROVED** [review]

**Pull request merged**